src/System.Net.Http/src/System/Net/Http/HttpClientHandler.Unix.cs

```diff
@@ -160,6 +161,10 @@ protected override void Dispose(bool disposing)
 
         protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
+            if (DiagnosticsHandler.IsEnabled())
```

Comment:
`EventSource` works well where it essentially evaporates if its not enabled (using the check pattern), but `DiagnosticSource` always pays the virtual dispatch cost event with an Enabled test